File: ui/src/main/java/com/peakmain/ui/recyclerview/group/GroupMultiType.java
package com.peakmain.ui.recyclerview.group;

import com.peakmain.ui.recyclerview.adapter.MultiTypeSupport;

/**
 * author ：Peakmain
 * createTime：2020/3/22
 * mail:2726449200@qq.com
 * describe：
 */
 class GroupMultiType<T extends GroupRecyclerBean> implements MultiTypeSupport<T> {
    private final int mGroupHeadResId;
    private final int mLayoutResId;

    public GroupMultiType(int layoutResId, int groupHeadResId) {
        this.mGroupHeadResId = groupHeadResId;
        this.mLayoutResId = layoutResId;
    }

    @Override
    public int getLayoutId(T item, int position) {
        if (item.isHeader) {
            return mGroupHeadResId;
        }
        return mLayoutResId;
    }

}

File: ui/src/main/java/com/peakmain/ui/recyclerview/group/GroupMultiType.kt
package com.peakmain.ui.recyclerview.group

import com.peakmain.ui.recyclerview.adapter.MultiTypeSupport

/**
 * author ：Peakmain
 * createTime：2020/3/22
 * mail:2726449200@qq.com
 * describe：
 */
internal class GroupMultiType<T : GroupRecyclerBean<T>?>(private val mLayoutResId: Int, private val mGroupHeadResId: Int) : MultiTypeSupport<T> {
    override fun getLayoutId(item: T, position: Int): Int {
        return if (item!!.isHeader) {
            mGroupHeadResId
        } else mLayoutResId
    }

}